django_project/feti/views/share.py

```diff
@@ -76,108 +81,111 @@ def download_map(self, filename, markers, provider=None):
-
-        template = get_template("feti/pdf_template.html")
-
-        if course_names:
-            query = ", ".join(course_names)
+        for result in data:
+            y, x = result['campus_location'].split(',')
+            markers += '%s,%s,ol-marker-blue|' % (y, x)
 
         context = {
             "type": "pdf",
-            "image": filename,
+            "image": map_image,
             "title": "Feti Report",
-            "provider": slug,
+            "provider": resource,
             "query": query,
-            "campuses": campuses
+            "campuses": data,
+            "options": self.advanced_search_options_to_string(options),
+            "url": self.feti_url(resource, self.request)
         }
 
+        self.download_map(filename=map_image, markers=markers, provider=resource)
+        template = get_template("feti/pdf_template.html")
+
         html = template.render(RequestContext(self.request, context))
+
         response = HttpResponse(content_type="application/pdf")
-        response['Content-Disposition'] = 'attachment; filename="%s.pdf"' % query
+        response['Content-Disposition'] = 'attachment; filename="feti.pdf"'
         weasyprint.HTML(string=html, url_fetcher=self.url_fetcher).write_pdf(response)
-
         return response
 
+    def url_fetcher(self, url):
+        if url.startswith('assets:'):
+            url = url[len('assets:'):]
+            url = "file://" + safe_join(settings.MEDIA_ROOT, url)
+        return weasyprint.default_url_fetcher(url)
+
+    def advanced_search_options_to_string(self, options):
+        # this should go somewhere else.
+        result = []
+        if 'fos' in options:
+            field_of_study_class = FieldOfStudy.objects.get(field_of_study_class=options['fos'])
+            result.append('Field of study: {}'.format(field_of_study_class))
+        if 'sos' in options:
+            subfield_of_study_class = SubFieldOfStudy.objects.get(id=int(options['sos']))
+            result.append('Subfield of study: {}'.format(subfield_of_study_class))
+        if 'qt' in options:
+            qualification_type = QualificationType.objects.get(id=int(options['qt']))
+            result.append('Qualification type: {}'.format(qualification_type))
+        if 'mc' in options:
+            result.append('Minimum credits: {}'.format(options['mc']))
+        if 'nqf' in options:
+            result.append('NQF level: {}'.format(options['nqf']))
+        if 'nqsf' in options:
+            result.append('NQSF level: {}'.format(options['nqsf']))
+        if 'pi' in options:
+            if options['pi'] == '1':
+                result.append('Public institution')
+            elif options['pi'] == '0':
+                result.append('Private institution')
+        return result
+
+    def feti_url(self, resource, request):
+        query_string = request.META['QUERY_STRING'].replace('export=pdf', '')
+        return "http://159.69.42.250:63100/#map/{}/{}".format(
```

Comment:
I think this shouldn't be hard coded

Reply:
You're right. But for the moment, it is hard-coded (for demo purpose)

Reply:
is it a must? is it using others url? 
i see it is just used as context in here : https://github.com/kartoza/feti/pull/584/files#diff-4140aa2bff85617f2e47c863f34d7cccR16

If it is not redirect to other url, you don't need this context, just use hostname function of javascript and add parameters from context after that

Reply:
Ok no problem, I removed the URL as this feature was not in the previous version.